test/integration/quota/quota_test.go

```diff
@@ -371,3 +388,208 @@ func TestQuotaLimitedResourceDenial(t *testing.T) {
 		t.Fatalf("unexpected error: %v", err)
 	}
 }
+
+// TestCRDQuotaUninstalledResources confirms that the quota controller continues to operate on CRDs that
+// are uninstalled and reinstalled on the cluster between quota controller syncs.
+func TestCRDQuotaUninstallReinstall(t *testing.T) {
+	ctx := setup(t, 2)
+	defer ctx.tearDown()
+	// install crd on cluster
+	ns := util.CreateNamespaceOrDie("test-crd", ctx.clientSet, t)
+	definition, resourceClient := util.CreateRandomCustomResourceDefinition(t, ctx.apiExtensionClient, ctx.dynamicClient, ns.Name)
+	time.Sleep(ctx.syncPeriod)
+	n := 3
+	quotaName := "quota"
+	hard := v1.ResourceList{}
+	resourceName := v1.ResourceName(fmt.Sprintf("count/%s.%s", definition.Spec.Names.Plural, definition.Spec.Group))
+	hard[resourceName] = resource.MustParse(strconv.Itoa(n))
+	quota := &v1.ResourceQuota{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      quotaName,
+			Namespace: ns.Name,
+		},
+		Spec: v1.ResourceQuotaSpec{
+			Hard: hard,
+		},
+	}
+	waitForQuota(t, quota, ctx.clientSet)
+	// create the crd instance
+	_, err := resourceClient.Create(context.TODO(), util.NewCRDInstance(definition, ns.Name, fmt.Sprintf("crd-%d", 1)), metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("failed to create crd instance 1, err: %v", err)
+	}
+
+	// confirm crd shows up in quota used count
+	err = pollExpectedQuota(ctx.clientSet, 1, resourceName, quotaName, ns.Name)
+	if err != nil {
+		t.Fatalf("err %v", err)
+	}
+
+	// delete the definition which should cascade to the instance and quota?
```

Comment:
no, quota should not be deleted, it is just not effective anymore

Reply:
fixed